src/Functions.Worker.Extensions.OpenAI/ChatMessage.cs

```diff
@@ -25,7 +27,7 @@ public ChatMessageEntity(string content, string role)
     public string Content { get; set; }
 
     /// <summary>
-    /// Gets or sets the role of the chat agent (“system”, “user”, "assistant", "function" or "tool").
+    /// Gets or sets the role of the chat agent.
     /// </summary>
-    public string Role { get; set; }
+    public ChatRole Role { get; set; }
```

Comment:
Does this work with the OOP chat bot scenario? I think it should just seralize but just wanted to validate :)

Reply:
Good question, let me check that. I might need to revert this change...

Reply:
Thanks for checking! I already have the work item on my plate. I'll start working on it as soon as we publish the nuget packages